### ISSUE TITLE
object-curly-spacing+objectsInObjects thinks `{ a: function () {} }` delimiter is an object.

### ISSUE THREAD

**Issue by Droogans** [eslint/eslint#5537]
**What version of ESLint are you using?**

v2.2.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**

I'm using ESLint to parse my source code.

**What did you do? Please include the actual source code causing the issue.**

When evaluating

```js
"object-curly-spacing": [2, "always", {
    "objectsInObjects": false
}]
```

I am being told that my no-op function in an inline object is an "objectInObject".

```js
var a = { noop: function () {} };
```

**What did you expect to happen?**

I expected ESLint to understand the difference between a closing bracket in a function block scope delimiter and a javascript object.

**What actually happened? Please include the actual, raw output from ESLint.**

It assumed the block scope delimiters were part of a javascript object.

**Comment by Droogans** [comment]
This may be related to https://github.com/eslint/eslint/issues/3763.

**Comment by nzakas** [comment]
Can you include the actual ESLint output?

**Comment by BYK** [comment]
This is related to #5538.

**Comment by BYK** [comment]
Verified on latest master.
